### PR TITLE
Train the model locally if there is only one worker using AllReduce strategy.

### PR DESCRIPTION
**Pull request by workingloong**

### CODE REVIEW

File: elasticdl/python/common/args.py [hunk 1]
```diff
@@ -200,6 +200,9 @@ def parse_worker_args(worker_args=None):
     parser.add_argument(
         "--worker_id", help="ID unique to the worker", type=int, required=True
     )
+    parser.add_argument(
+        "--num_workers", help="The number of workers", type=int, required=True
```
Comment (terrytangyuan) [inline elasticdl/python/common/args.py]: This new required arg will break compatibility. Is this intended?

Comment (workingloong) [reply elasticdl/python/common/args.py]: Yes. If `num_workers=1`, we cannot use `CollectiveCommunicator` which cannot work if only a worker is configured. We can train the model locally without `CollectiveCommunicator` with a worker.

Comment (terrytangyuan) [reply elasticdl/python/common/args.py]: Yes but you are adding a new required argument here in common args, which will potentially break existing usage.

Comment (skydoorkai) [reply elasticdl/python/common/args.py]: Workers are launched by the master. 
The master will always have`--num_workers` argument which will be passed to workers.

Thus, nothing will be broken.

Comment (skydoorkai) [reply elasticdl/python/common/args.py]: But need to add the argument in the unittest.

Comment (terrytangyuan) [reply elasticdl/python/common/args.py]: Ok

**Review by terrytangyuan — DISMISSED** [review]

**Review by skydoorkai — APPROVED** [review]

**Pull request merged**